modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/SystemObjectEntry1toMObjectRelationshipElementsParserImpl.java

```diff
@@ -47,6 +48,10 @@ public List<Map<String, Object>> parse(
 		if (objectRelationship.getObjectDefinitionId1() ==
 				objectDefinition.getObjectDefinitionId()) {
 
+			if (MapUtil.isEmpty(parseOne(value))) {
```

Comment:
You are invoking the parseOne method twice. Maybe better to keep the result in a variable